Mathlib/Analysis/InnerProductSpace/ProdL2.lean

```diff
@@ -14,11 +14,52 @@ The `L²` norm on product of two inner product spaces is compatible with an inne
 $$
 \langle x, y\rangle = \langle x_1, y_1 \rangle + \langle x_2, y_2 \rangle.
 $$
-This is recorded in this file as an inner product space instance on `ProdLp 2`.
+This is recorded in this file as an inner product space instance on `WithLp 2 (E × F)`.
 -/
 
+local macro_rules | `($x ^ $y)   => `(HPow.hPow $x $y) -- Porting note: See issue lean4#2220
+
+open Real ENNReal NNReal
+
 namespace WithLp
 
+section norm_of
+
+variable {α β : Type*}
+variable [SeminormedAddCommGroup α] [SeminormedAddCommGroup β]
+
+theorem prod_norm_eq_of_L2 (x : WithLp 2 (α × β)) : ‖x‖ = sqrt (‖x.fst‖ ^ 2 + ‖x.snd‖ ^ 2) := by
+  rw [prod_norm_eq_of_nat 2 (by norm_cast) _, Real.sqrt_eq_rpow]
+  norm_cast
+
+theorem prod_nnnorm_eq_of_L2 (x : WithLp 2 (α × β)) :
+    ‖x‖₊ = NNReal.sqrt (‖x.fst‖₊ ^ 2 + ‖x.snd‖₊ ^ 2) :=
+  NNReal.eq <| by
+    push_cast
+    exact prod_norm_eq_of_L2 x
+
+theorem prod_norm_sq_eq_of_L2 (x : WithLp 2 (α × β)) : ‖x‖ ^ 2 = ‖x.fst‖ ^ 2 + ‖x.snd‖ ^ 2 := by
+  suffices ‖x‖₊ ^ 2 = ‖x.fst‖₊ ^ 2 + ‖x.snd‖₊ ^ 2 by
+    simpa only [NNReal.coe_sum] using congr_arg ((↑) : ℝ≥0 → ℝ) this
+  rw [prod_nnnorm_eq_of_L2, NNReal.sq_sqrt]
+
+theorem prod_dist_eq_of_L2 (x y : WithLp 2 (α × β)) :
+    dist x y = (dist x.fst y.fst ^ 2 + dist x.snd y.snd ^ 2).sqrt := by
+  simp_rw [dist_eq_norm, prod_norm_eq_of_L2]
+  rfl
+
+theorem prod_nndist_eq_of_L2 (x y : WithLp 2 (α × β)) :
+    nndist x y = NNReal.sqrt (nndist x.fst y.fst ^ 2 + nndist x.snd y.snd ^ 2) :=
+  NNReal.eq <| by
+    push_cast
+    exact prod_dist_eq_of_L2 _ _
+
+theorem prod_edist_eq_of_L2 (x y : WithLp 2 (α × β)) :
+    edist x y = (edist x.fst y.fst ^ 2 + edist x.snd y.snd ^ 2) ^ (1 / 2 : ℝ) := by
+  simp [prod_edist_eq_add]
+
+end norm_of
```

Comment:
I don't think I agree with moving these lemmas; it means that these lemmas come with the extra import requirements of `InnerProductSpace`, which I suspect is reasonable heavy due to it requiring `IsROrC`.

Reply:
@eric-wieser : are you referring to this comment?

> Also, the `L^2` statements would probably fit better in the `ProdL2` file.

If so, agreed. Even though they are the specializations to `p = 2`, they belong in the `ProdLp` file.